tools/common/__init__.py

```diff
@@ -192,7 +192,8 @@ async def update_outbound_ws(
 async def wait_for_keyboard_interrupt() -> None:
     """Wait for keyboard interrupt (Ctrl-C)."""
     sig_event = asyncio.Event()
-    signal.signal(signal.SIGINT, lambda _exit_code, _frame: sig_event.set())
+    loop = asyncio.get_running_loop()
+    loop.add_signal_handler(signal.SIGINT, lambda: sig_event.set())
```

Comment:
I assume you use a UNIX based environment, these methods are only supported on Unix:
https://peps.python.org/pep-3156/#signal-callbacks
<img width="360" height="64" alt="image" src="https://github.com/user-attachments/assets/201bd66d-063f-4959-81a7-3dc7abe986ec" />

I have also tested it on Windows and it does not work, I don't think it worth changing this as many users use this to collect data or test devices on Windows.



```suggestion
    signal.signal(signal.SIGINT, lambda _exit_code, _frame: sig_event.set())
```